homeassistant/components/laundrify/binary_sensor.py

```diff
@@ -0,0 +1,73 @@
+"""Platform for binary sensor integration."""
+from __future__ import annotations
+
+import logging
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, MANUFACTURER, MODEL
+from .coordinator import LaundrifyUpdateCoordinator
+from .model import LaundrifyDevice
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, config: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up sensors from a config entry created in the integrations UI."""
+
+    coordinator = hass.data[DOMAIN][config.entry_id]["coordinator"]
+
+    async_add_entities(
+        LaundrifyPowerPlug(coordinator, device) for device in coordinator.data.values()
+    )
+
+
+class LaundrifyPowerPlug(
+    CoordinatorEntity[LaundrifyUpdateCoordinator], BinarySensorEntity
+):
+    """Representation of a laundrify Power Plug."""
+
+    _attr_device_class = BinarySensorDeviceClass.RUNNING
+    _attr_icon = "mdi:washing-machine"
+
+    coordinator: LaundrifyUpdateCoordinator
+
+    def __init__(
+        self, coordinator: LaundrifyUpdateCoordinator, device: LaundrifyDevice
+    ) -> None:
+        """Pass coordinator to CoordinatorEntity."""
+        super().__init__(coordinator)
+        self._device = device
+        self._attr_unique_id = device["_id"]
+        self._attr_name = device["name"]
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Configure the Device of this Entity."""
+        return DeviceInfo(
+            identifiers={(DOMAIN, self._device["_id"])},
+            name=self.name,
+            manufacturer=MANUFACTURER,
+            model=MODEL,
+            sw_version=self._device["firmwareVersion"],
+        )
+
+    @property
+    def available(self) -> bool:
+        """Check if the device is available."""
+        return self.unique_id in self.coordinator.data
+
+    @property
+    def is_on(self) -> bool:
+        """Return entity state."""
+        return bool(self.coordinator.data[self.unique_id]["status"] == "ON")
```

Comment:
I actually had to cast explicitly here, since `mypy` would throw an error otherwise:

```
root ➜ /workspaces/core (laundrify ✗) $ mypy homeassistant/components/laundrify

homeassistant/components/laundrify/binary_sensor.py:71: error: Returning Any from function declared to return "bool"  [no-any-return]

Found 1 error in 1 file (checked 6 source files)
```

Let me know if there's another solution to this.

Reply:
https://github.com/home-assistant/core/pull/65090#discussion_r870415134 this should resolve the typing issue. Give it a shot!

Reply:
Changing the return typehint as proposed wasn't enough for some reason. Turned out it can be resolved by accessing the internal `_device` property instead of the raw data of the coordinator :)

Fixed in 831beaaf286d3d408750791d7fdf22d32a4a00cc